scala/scala.js

```diff
@@ -0,0 +1,1270 @@
+      var bufferId = instance.find('.jsvee-variable[data-name="items"] .jsvee-value').first().attr('data-id');
+      var bufferInstance = area.find('.jsvee-heap .jsvee-instance[data-id="' + bufferId + '"]').first();
+      bufferInstance.append($(element.find('.jsvee-value').get(1)).clone());
+      ready();
+    }
+  };
+
+  JSVEE.handlers.classes['Array'] = function (ready, area, element) {
+    if (element.hasClass("jsvee-function") && element.attr('data-name') == 'Array') {
+      createArrayBuffer(ready, area, element, "Array");
+    } else if (element.attr('data-name') === "( )") {
+      var prev = element.prev();
+      var index = parseInt(element.find('.jsvee-value').data('value'));
+      var buffer = area.find('.jsvee-heap .jsvee-instance[data-id="' + prev.data('id') + '"]').first();
+      var item = $(buffer).find('.jsvee-item').get(index);
+      prev.remove();
+      ready($(item).find('.jsvee-value').first().clone());
+    } else if (element.attr('data-name') === "( ) =") {
+      var prev = element.prev();
+      var next = element.next();
+      var index = parseInt(element.find('.jsvee-value').data('value'));
+      var buffer = area.find('.jsvee-heap .jsvee-instance[data-id="' + prev.data('id') + '"]').first();
+      var item = $(buffer).find('.jsvee-item').get(index);
+      $(item).find('.jsvee-value').replaceWith(next.clone());
+      prev.remove();
+      next.remove();
+      ready();
+    }
+  };
+
+  var createArrayBuffer = function (ready, area, element, type) {
+    var instance = JSVEE.utils.ui.createInstance(area, type);
+    area.find('.jsvee-heap').append(instance);
+
+    var bufferElement = instance;
+    $(element).find('.jsvee-value').each(function (index, value) {
+      var item = $('<div></div>').addClass('jsvee-item');
+      item.css('display', 'inline-block');
+      item.appendTo(bufferElement);
+      $(value).clone().appendTo(item);
+    });
+
+    ready(JSVEE.utils.ui.createReference(area, instance.attr('data-id')));
+  };
+
+  var createUnit = function () {
+    var val = $('<div>Unit</div>').addClass('jsvee-value');
+    val.attr('data-type', 'Unit').attr('data-value', 'Unit');
+    val.attr('title', 'Unit, tyhjä palautusarvo');
+    return val;
+  };
+
+  JSVEE.ui.finishLayout = function () {
+
+    if (this.engine.settings.noIcons) {
+      this.area.find('.jsvee-class.jsvee-instance').each(function () {
+        var elem = $(this);
+        elem.attr('title', '');
+        elem.find('.jsvee-name').text(elem.find('.jsvee-name2').text());
+      });
+    }
+
+    modifyUI.call(this.engine);
+
+  };
+
+  JSVEE.handlers.explanations.addOperator = function (op, position) {
+    var type = "";
+    if (position === undefined) {
+      type = this.area.find(".jsvee-evaluation-area").first().children().last().data('type');
+    } else {
+      type = JSVEE.utils.ui.findElement(this.area, position, true).data('type');
+    }
+
+    var name = op + "-operaattori";

```

Comment:
Not sure how translations should be handled here. This aspect needs to be designed and developed as current main user, Aalto Ohjelmointi 1. requires support for two Languages.